### PR TITLE
Add nightly build and failure notification on `kedro-starters`

### PR DESCRIPTION
**Pull request by ankatiyar**
## Checklist

- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Assigned myself to the PR
- [ ] Added tests to cover my changes

### CODE REVIEW

**Review by ElenaKhaustova — APPROVED** [review]
Thank you, @ankatiyar, LGTM!

The only issue is with `python3.8` tests because of moving the transcoding logic. But I think we can ignore it in the context of current PR.

File: .github/workflows/all-checks.yml [hunk 1]
```diff
@@ -7,6 +7,8 @@ on:
   pull_request:
     branches:
       - main
+  workflow_dispatch:
+  workflow_call:
```
Comment (merelcht) [inline .github/workflows/all-checks.yml:11]: Does this do anything?

Comment (ankatiyar) [reply .github/workflows/all-checks.yml:11]: `workflow_dispatch` makes it possible to trigger it manually from the Actions tab and `workflow_call` makes it possible for nightly-build to call this. I could move `workflow_dispatch` from here to nightly build actually!

Comment (ankatiyar) [inline .github/workflows/all-checks.yml:11]:
```suggestion
  workflow_call:
```

**Review by merelcht — APPROVED** [review]

**Pull request merged**